src/System.Web.OData/OData/Query/Expressions/PropertyContainer.cs

```diff
@@ -54,29 +56,44 @@ protected PropertyContainer()
         //          Name = properties[0].Key, 
         //          Value = properties[0].Value,
         //
-        //          Next = new NamedProperty<> { ..... } 
+        //          LeftNext = new NamedProperty<> { ..... } 
+        //          RightNext = new NamedProperty<> { ..... } 
         //      }
         public static Expression CreatePropertyContainer(IList<NamedPropertyExpression> properties)
         {
             Expression container = null;
 
             // build the linked list of properties.
-            foreach (NamedPropertyExpression property in properties)
+            if (properties.Count >= 1)
             {
-                container = CreateNamedPropertyCreationExpression(property, container);
+                NamedPropertyExpression property = properties.First();
+                int count = properties.Count - 1;
+                int leftSize = GetLeftSize(count);
+                Expression leftNext = CreatePropertyContainer(properties.Skip(1).Take(leftSize).ToList());
+                Expression rightNext = CreatePropertyContainer(properties.Skip(1 + leftSize).ToList());
+                container = CreateNamedPropertyCreationExpression(property, leftNext, rightNext);
             }
 
             return container;
         }
 
+        private static int GetLeftSize(int count)
```

Comment:
OData repo doesn't support C# 6.0 yet :(. I can't use that syntax

---
In reply to: [131731004](https://github.com/kosinsky/WebApi/pull/9#discussion_r131731004) [](ancestors = 131731004)